libraries/AP_Motors/examples/AP_Motors_test/AP_Motors_test.cpp

```diff
@@ -30,105 +27,175 @@ void stability_test();
+                // we have access to certain functions in the multicopter motors child class
+                if (i != 2) {
+                    ::printf("frame_class must be second argument\n");
+                    exit(1);
+                }
+
+                // Setup the correct motors object for the frame class to test
+                frame_class = (AP_Motors::motor_frame_class)value;
+
+                switch (frame_class) {
+
+                    case AP_Motors::MOTOR_FRAME_QUAD:
+                    case AP_Motors::MOTOR_FRAME_HEXA:
+                    case AP_Motors::MOTOR_FRAME_OCTA:
+                        motors_matrix = new AP_MotorsMatrix(400);
+                        motors = motors_matrix;
+                        num_outputs = __builtin_popcount(motors->get_motor_mask());
+                        break;
+
+                    case AP_Motors::MOTOR_FRAME_HELI:
+                        motors = new AP_MotorsHeli_Single(400);
+                        num_outputs = 6; // Mot 1-3 swashplate, mot 4 tail rotor pitch, mot 6 for 4th servo in H4-90 swash
+                        break;
+
+                    case AP_Motors::MOTOR_FRAME_HELI_DUAL:
+                        motors = new AP_MotorsHeli_Dual(400);
+                        num_outputs = 8; // Mot 1-3 swashplate 1, mot 4-5 swashplate 2, mot 6 and 7 for 4th servos on H4-90 swash plates front and back, respectively
+                        break;
+
+                    case AP_Motors::MOTOR_FRAME_HELI_QUAD:
+                        motors = new AP_MotorsHeli_Quad(400);
+                        num_outputs = 4; // only 4 collective servos
+                        break;
+
+                    default:
+                        ::printf("ERROR: frame_class=%d not implemented\n", frame_class);
+                        exit(1);
+                }
+
+                // Init motors
+                motors->init(frame_class, AP_Motors::MOTOR_FRAME_TYPE_X);
+
+                // Check that init worked
+                if (!motors->initialised_ok()) {
+                    ::printf("ERROR: frame_class=%d initialisation failed\n", frame_class);
+                    exit(1);
+                }
+
+                motors->output_min();
+
             } else {
-                ::printf("Expected \"yaw_headroom\" or \"throttle_avg_max\"\n");
+                ::printf("Expected \"frame_class\", \"yaw_headroom\" or \"throttle_avg_max\"\n");
                 exit(1);
             }
+        }
 
+        // if we haven't been given a frame class by argument we just assume a quad by default
+        // so that the single first argument s or t still works
+        if (motors == nullptr) {
+            motors_matrix = new AP_MotorsMatrix(400);
+            motors = motors_matrix;
+            motors->init(AP_Motors::MOTOR_FRAME_QUAD, AP_Motors::MOTOR_FRAME_TYPE_X);
+            num_outputs = 4;
         }
-#endif
+
```

Comment:
```suggestion
        motors->set_dt(1.0/400.0);
        motors->set_update_rate(490);
        motors->output_min();
```

Reply:
@IamPete1 Not sure if this affects what you are doing with this test but Heli's don't have update rates of 490.  If this is referring to the RC_SPEED then we set it to 125.

Reply:
The 490 refers to AP_MOTORS_SPEED_DEFAULT https://github.com/ArduPilot/ardupilot/blob/1f9539a1b3416492c24e3ef73d4570ed5c3f9f4c/libraries/AP_Motors/AP_Motors_Class.h#L52  I do not know the history of this number, but it gets overridden by all motors classes in copter to be copter.scheduler.get_loop_rate_hz().

Update rates don't actually effect the results of the motor test, because we run each case 1000 times to remove slew rates and filters etc from the result.  So I think this is fine as it is. 
https://github.com/ArduPilot/ardupilot/blob/1f9539a1b3416492c24e3ef73d4570ed5c3f9f4c/libraries/AP_Motors/examples/AP_Motors_test/AP_Motors_test.cpp#L337-L340

Reply:
This update rate is the rate of the PWM outputs, it actually clamps down to the loop rate so on copter you get a 1000us to 2000us pulse at 400hz. As Bill says on heli it would be at 125hz because servos sometimes don't like the faster updates. However its related to how we generate the PWM signal and output it, so it makes no difference for the purposes of this test.